I.write-tests/05-get-second-third-smallest/get-second-third.test.js

```diff
@@ -1,5 +1,16 @@
 // example
 // input = [90, 5, 11, 8, 6];
 // expected output = [6, 8];
+const getSecondThird = require('./get-second-third');
+let input = [90, 5, 11, 8, 6];
+const inputLenInital = input.length;
+let result = getSecondThird(input);
+const inputLenAfterPassed = input.length;
 
-// also test that the original array has not changed
\ No newline at end of file
+test('should get second and third higerst numbers', () =>{
+    expect(result).toEqual([6, 8]);
+});
+// also test that the original array has not changed
+test('should not change original input array', () => {
+    expect(inputLenAfterPassed).toBe(inputLenInital);
```

Comment:
Unfortunately this test doesn't quite do what you want it to. For example the implementation code may mutate the array that it receives as a parameter, so just checking the length isn't enough unfortunately.

To check that an array is in the same order as it was before, you would have to do something like:

```js
expect(input).toEqual([90, 5, 11, 8, 6])
```

I know this looks a bit strange as you would think that you could just replace the array here with the input variable. However due to the way *memory* works in JS, `input` might not be the same as that array after the test has run.

Reply:
Fixed as above!

Reply:
Just to be clear - your test here could be wrong still. For example, if the implementation code re-assigned the array that it was given to be an array of the same length but filled it with random numbers, then this test would still pass. The length of an array is not enough to check that it's contents are what you expect.

Reply:
ok